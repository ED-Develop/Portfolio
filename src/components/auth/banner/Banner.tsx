import React, {CSSProperties} from 'react';
import style from '../Auth.module.scss';
import logo from '../../../assets/images/logo-light.png';
import {Layout} from 'antd';
import banner from '../../../assets/images/login-banner.jpg';
import {NavLink} from 'react-router-dom';

const {Content} = Layout;

export const Banner = () => {
    const styles: CSSProperties = {
        backgroundImage: `linear-gradient(rgba(0, 0, 0, 0.2), rgba(0, 0, 0, 0.2)), url(${banner})`
    };

    const links = [
        {path: '/', title: 'About'},
        {path: '/', title: 'Contact'},
    ];

    return (
        <Content className={style.landing__banner} style={styles}>
                <div className={style.logo}>
                    <img src={logo} alt="logo"/>
                </div>
                <div className={style.title}>
                    <h1>Make your life more interesting with Socialite Social Network</h1>
                </div>
                <ul className={style.nav}>
                    {
                        links.map(link => (
                            <li key={link.title}>
                                <NavLink to={link.path}>{link.title}</NavLink>
                            </li>
                        ))
                    }
                </ul>
        </Content>
    )
};
