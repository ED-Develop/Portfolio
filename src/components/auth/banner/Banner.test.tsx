import React from 'react';
import {render} from '@testing-library/react';
import {Banner} from './Banner';
import {BrowserRouter} from 'react-router-dom';

describe('Banner component: ', () => {
    test('Should renders correctly', () => {
        const {container} = render(
            <BrowserRouter>
                <Banner/>
            </BrowserRouter>
        );

        expect(container).toMatchSnapshot();
    });
});