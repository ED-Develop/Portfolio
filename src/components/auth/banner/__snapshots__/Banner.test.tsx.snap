// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Banner component:  Should renders correctly 1`] = `
<div>
  <main
    class="ant-layout-content landing__banner"
  >
    <div
      class="logo"
    >
      <img
        alt="logo"
        src="logo-dark.png"
      />
    </div>
    <div
      class="title"
    >
      <h1>
        Make your life more interesting with Socialite Social Network
      </h1>
    </div>
    <ul
      class="nav"
    >
      <li>
        <a
          aria-current="page"
          class="active"
          href="/"
        >
          About
        </a>
      </li>
      <li>
        <a
          aria-current="page"
          class="active"
          href="/"
        >
          Contact
        </a>
      </li>
    </ul>
  </main>
</div>
`;
